apps/frontend/src/views/Checklist.vue

```diff
@@ -0,0 +1,313 @@
+import IconLinkItem from '@/components/global/sidebaritems/IconLinkItem.vue';
+import {AppInfoModule} from '@/store/app_info';
+import ChecklistRulesTable from '@/components/global/checklist/ChecklistRulesTable.vue';
+import ChecklistRuleInfoHeader from '@/components/global/checklist/ChecklistRuleInfoHeader.vue';
+import ChecklistRuleInfoBody from '@/components/global/checklist/ChecklistRuleInfoBody.vue';
+import ChecklistRuleEdit from '@/components/global/checklist/ChecklistRuleEdit.vue';
+import ChecklistSeverityOverride from '@/components/global/checklist/ChecklistSeverityOverride.vue';
+import ExportButton from '@/components/generic/ExportButton.vue';
+
+@Component({
+  components: {
+    Base,
+    UploadButton,
+    IconLinkItem,
+    ChecklistRulesTable,
+    ChecklistRuleInfoHeader,
+    ChecklistRuleInfoBody,
+    ChecklistRuleEdit,
+    ChecklistSeverityOverride,
+    ExportButton
+  }
+})
+export default class Checklist extends RouteMixin {
+  /** Model for if all-filtered snackbar should be showing */
+  filterSnackbar = false;
+  controlSelection: string | null = null;
+
+  /** State variable to track status of bottom-sheet */
+  sheet = false;
+
+  /** State variable to track "Short ID" switch */
+  shortIdEnabled = true;
+
+  //** Variable for selected tab */
+  tab = null;
+
+  evalInfo:
+    | SourcedContextualizedEvaluation
+    | SourcedContextualizedProfile
+    | null = null;
+
+  exportCkl() {
+    type FileData = {
+      filename: string;
+      data: string;
+    };
+    const checklist = this.getChecklist(this.fileFilter);
+    if (checklist) {
+      const checklistString = ChecklistConverter.toChecklist(checklist);
+      const file: FileData[] = [
+        {
+          filename: checklist.filename,
+          data: '<?xml version="1.0" encoding="UTF-8"?>' + checklistString
+        }
+      ];
+      saveSingleOrMultipleFiles(file, 'ckl');
+    }
+  }
+
+  get selectedRule() {
+    let stillExists = false;
+    // Checks to see if the selected rule still exists after filtering
+    this.rules.forEach((item) => {
+      if (_.isEqual(FilteredDataModule.selectedRule, item)) {
+        stillExists = true;
+      }
+    });
+    if (stillExists) {
+      return FilteredDataModule.selectedRule;
+    }
+    return this.rules[0] ?? FilteredDataModule.emptyRule;
+  }
+
+  /**
+   * The current search terms, as modeled by the search bar
+   */
+  get searchTerm(): string {
+    return SearchModule.searchTerm;
+  }
+
+  set searchTerm(term: string) {
+    SearchModule.updateSearch(term);
+  }
+
+  /**
+   * The currently selected file, if one exists.
+   * Controlled by router.
+   */
+  get fileFilter(): FileID[] {
+    return FilteredDataModule.selectedChecklistIds;
+  }
+
+  getChecklist(fileID: FileID[]) {
+    return InspecDataModule.allChecklistFiles.find(
+      (f) => f.uniqueId === fileID[0]
+    );
+  }
+
+  /**
+   * The filter for charts. Contains all of our filter stuff
```

Comment:
comment doesn't make sense

Reply:
Made a detailed comment, removing the out-dated one